plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1087,29 +1087,133 @@ public void testCaseInsensitivePredicate()
     }
 
     @Test
-    public void testArithmeticModuloPredicatePushdown()
+    public void testArithmeticPredicatePushdown()
     {
         if (!hasBehavior(SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN)) {
-            assertThat(query("SELECT shippriority FROM orders WHERE shippriority % 4 = 0")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + 1 = 7"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 1 = 7"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey * 1 = 7"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey / 1 = 7"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey % 20 = 7"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE -(nationkey) = -7"))
+                    .isNotFullyPushedDown(FilterNode.class);
             return;
         }
-        assertThat(query("SELECT shippriority FROM orders WHERE shippriority % 4 = 0")).isFullyPushedDown();
+        // negate
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE -(nationkey) = -7"))
+                .matches("VALUES (BIGINT '7', CAST('GERMANY' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE -(nationkey) = 0"))
+                .matches("VALUES (BIGINT '0', CAST('ALGERIA' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
 
-        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % nationkey = 2"))
```

Comment:
@findepi Since you originally added these in https://github.com/trinodb/trino/pull/11510/files#diff-01f7e5fb80c1ce75cf0ea5d3f1933156eb7984967b99b92090bf931e93569696R711:

Is the usage of a tuple-domain (`>`) ANDed to complex expression significant in any way?
Also is `(nationkey - regionkey) % nationkey` providing some specific coverage compared to say `nationkey % regionkey`?
Or did you do that to cover subtraction as well in same test?

Reply:
> Or did you do that to cover subtraction as well in same test?

This